eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/AsyncHttpRequestManager.java

```diff
@@ -70,14 +111,43 @@ public static void initialize(boolean isDebugEnabled) {
     if (isNotInitialized()) {
       LOGGER.info("Initializing Asynchronous HTTP manager...");
 
-      // Initialize the asynchronous HTTP client
-      RequestConfig requestConfig =
-          RequestConfig.custom()
-              .setConnectionRequestTimeout(CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
-              .setResponseTimeout(SOCKET_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
-              .build();
-      httpAsyncClient = HttpAsyncClients.custom().setDefaultRequestConfig(requestConfig).build();
-      httpAsyncClient.start();
+      // Create connection manager
+      try {
+        asyncClientConnectionManager = new PoolingAsyncClientConnectionManager();
+        asyncClientConnectionManager.setMaxTotal(HTTP_CXN_POOL_MAX_REQUESTS);
+        asyncClientConnectionManager.setDefaultMaxPerRoute(HTTP_CXN_POOL_MAX_REQUESTS_PER_ROUTE);
+
+        // Create IO reactor configuration
+        IOReactorConfig ioReactorConfig =
+            IOReactorConfig.custom()
+                .setIoThreadCount(Runtime.getRuntime().availableProcessors() * 2)
+                .setSoTimeout(SOCKET_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
+                .setSoKeepAlive(false)
+                .build();
+
+        // Create connection configuration
+        ConnectionConfig connectionConfig =
+            ConnectionConfig.custom()
+                .setConnectTimeout(CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
+                .build();
+        asyncClientConnectionManager.setDefaultConnectionConfig(connectionConfig);
+
+        // Initialize the asynchronous HTTP client
+        RequestConfig requestConfig =
+            RequestConfig.custom()
+                .setConnectionRequestTimeout(CONNECTION_REQUEST_TIMEOUT_SECS, TimeUnit.SECONDS)
+                .setResponseTimeout(RESPONSE_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
+                .build();
+        httpAsyncClient =
+            HttpAsyncClients.custom()
+                .setDefaultRequestConfig(requestConfig)
+                .setIOReactorConfig(ioReactorConfig)
+                .setConnectionManager(asyncClientConnectionManager)
+                .build();
+        httpAsyncClient.start();
+      } catch (Exception e) {
+        LOGGER.error("Failed to create HTTP connection manager.", e);
```

Comment:
Should application continue if there is an error here?

Reply:
Might as well exit -- Adding a return value and using it to update the startup success flag